crates/uv/src/commands/pip_sync.rs

```diff
@@ -291,7 +294,13 @@ pub(crate) async fn pip_sync(
             .dimmed()
         )?;
 
-        resolution.into_distributions().collect::<Vec<_>>()
+        resolution
+            .into_distributions()
+            .filter_map(|dist| match dist {
+                ResolvedDist::Installable(dist) => Some(dist),
+                ResolvedDist::Installed(_) => None,
```

Comment:
Nice, and these just get left as-is.

Reply:
I'm trying to think through how this intersects with `reinstalls` though... Like, what happens if we end up having to use a different version than one of the installed versions? How will the existing version be uninstalled? Would it already have been included in reinstalls somehow...?

Reply:
Yeah, I think they would in theory already be included in `reinstalls`... Does that rely on the assumption that we pass exact-equality requirements into the `Planner`? We might already rely on this today in `pip install`?

Reply:
Like the problem I'm imagining is something like... We have `foo==2.0.0` installed. The user requests `foo`. `foo==2.0.0` satisfies `foo`, but then we go to resolve, and we find that we need `foo==3.0.0`. But we'd only learn that when resolving, not when creating the install plan...

It shouldn't matter here since it's not a "real" resolution, we're just looking for a distribution for each version. It could matter for `pip install`, but in that case we resolve first, then use the `requirements` from the resolution as the inputs to the plan, which are pinned to exact versions. Anyway, there's probably some (existing) implicit coupling there that we should consider making explicit.

Reply:
Since #2696 was merged I'm just telling the resolver that there are no installed packages available for `pip sync`. We can consider how we want to handle that separately?

Do your comments apply outside of this command?

Reply:
I don't think my comments merit any changes to the PR. I'm more commenting on the fact that we pass `requirements` to `Plan` to tell it what versions we're going to need, but a `Requirement` is actually pretty loose.

Reply:
Yeah I noticed that too, like I kind of wanted the `ResolvedDist` here since the resolver _knows_ an installed version should be used already. I think there's probably room for future clean-up here.